mismi-s3/src/Mismi/S3/Commands.hs

```diff
@@ -520,6 +533,10 @@ handle404 :: AWS a -> AWS (Maybe a)
 handle404 m = fmap Just m `catch` \ (e :: Error) ->
   if e ^? httpStatus == Just status404 then return Nothing else throwM e
 
+handle301 :: AWS a -> AWS (Maybe a)
+handle301 m = fmap Just m `catch` \ (e :: Error) ->
+  if e ^? httpStatus == Just status301 then return Nothing else throwM e
+
```

Comment:
Yeh, I think this needs to be talked through a bit. @nhibberd Had an issue raised somewhere about handling the 301s, I think it is more subtle than it looks, but you can get 301 instead of 404 if you request something in a real public bucket instead an entirely missing one. This is annoying, but possibly unavoidable without breaking 301 behaviour in the actual incorrect location case. Probably worth a whiteboard on monday.